app/vue/src/client/preview/types-6-0.ts

```diff
@@ -0,0 +1,23 @@
+import { Component } from 'vue';
+import { Args as DefaultArgs, Annotations, BaseMeta, BaseStory } from '@storybook/addons';
+import { StoryFnVueReturnType } from './types';
+
+export { Args, ArgTypes, Parameters, StoryContext } from '@storybook/addons';
+
+type VueComponent = any;
```

Comment:
Did you mean `Component`? deepscan is giving a warning about that import

Reply:
I did meant any (but forgot to cleanup my tests using Component leading to unused import)

This is only a very first step of types for Vue, it's not infering args as I would expect but no idea how to make this working for now. Need Vue experts :)

Reply:
@backbone87 @Aaron-Pool any thoughts on the Component type in Vue?

Reply:
To represent a general component, `Component<any, any, any, any> | AsyncComponent<any, any, any, any>`, which is the typing for `components` property of components, would be good IMO. (or refer it with `ComponentOptions<any>['components']['string']` ...? 🤔 )

https://github.com/vuejs/vue/blob/dev/types/options.d.ts#L102